### PR TITLE
spiutils: Add firmware protocol.

### PR DESCRIPTION
**Pull request by osenft**
This change adds the protocol to update firmware.

### CODE REVIEW

File: shared-lib/spiutils/src/protocol/firmware.rs [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+    fn to_wire<W: Write>(&self, mut w: W) -> Result<(), ToWireError> {
+        w.write_be(self.content.to_wire_value())?;
+        Ok(())
+    }
+}
+
+// ----------------------------------------------------------------------------
+
+wire_enum! {
+    /// The content type.
```
Comment (dann21) [inline shared-lib/spiutils/src/protocol/firmware.rs]: This comment looks like a remnant.

File: shared-lib/spiutils/src/protocol/firmware.rs [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+use crate::io::Read;
+use crate::io::Write;
+use crate::protocol::wire::FromWireError;
+use crate::protocol::wire::FromWire;
+use crate::protocol::wire::ToWireError;
+use crate::protocol::wire::ToWire;
+use crate::protocol::wire::WireEnum;
+
+wire_enum! {
+    /// The content type.
```
Comment (dann21) [inline shared-lib/spiutils/src/protocol/firmware.rs:28]: Why do all the comments in this wire-enum have three slashes?

File: shared-lib/spiutils/src/protocol/firmware.rs [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+        w.write_be(self.segment_and_location.to_wire_value())?;
+        w.write_be(self.offset)?;
+        Ok(())
+    }
+}
+
+// ----------------------------------------------------------------------------
+
+wire_enum! {
+    /// The result of an update prepare request.
```
Comment (dann21) [inline shared-lib/spiutils/src/protocol/firmware.rs]: Is this comment right?

File: shared-lib/spiutils/src/protocol/firmware.rs [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+
+        /// Too much data
+        DataTooLong = 0x04,
+
+        /// Post-write compare failed
+        CompareFailed = 0x05,
+    }
+}
+
+/// A parsed update prepare response.
```
Comment (dann21) [inline shared-lib/spiutils/src/protocol/firmware.rs]: Is this comment right?

**Review by dann21 — APPROVED** [review]
Several nits, but otherwise LGTM.

File: shared-lib/spiutils/src/protocol/firmware.rs (hunk 2 above)
Comment (osenft) [reply shared-lib/spiutils/src/protocol/firmware.rs:28]: The `///` indicates that this should be included in documentation.

File: shared-lib/spiutils/src/protocol/firmware.rs (hunk 4 above)
Comment (osenft) [reply shared-lib/spiutils/src/protocol/firmware.rs]: Fixed. Sorry for that.

File: shared-lib/spiutils/src/protocol/firmware.rs (hunk 3 above)
Comment (osenft) [reply shared-lib/spiutils/src/protocol/firmware.rs]: Fixed. Sorry for that.

File: shared-lib/spiutils/src/protocol/firmware.rs (hunk 1 above)
Comment (osenft) [reply shared-lib/spiutils/src/protocol/firmware.rs]: Fixed. Sorry for that.

**Review by dann21 — APPROVED** [review]
LGTM.

**Review by dann21 — APPROVED** [review]

**Pull request merged**